pkg/test/util/structpath/structpath.go

```diff
@@ -0,0 +1,236 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package structpath
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"gopkg.in/d4l3k/messagediff.v1"
+	"reflect"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/gogo/protobuf/jsonpb"
+	"github.com/golang/protobuf/proto"
```

Comment:
Mixing golang/proto and gogo/proto may lead to unexpected results. Maybe use gogo consistently?